src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +224,871 @@ static DamageTable D_8092C140 = {
+        Math_ScaledStepToS(&this->actor.shape.rot.y, temp_a1, 0x4B0);
+    } else {
+        Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x4B0);
+    }
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if (EnIk_IsChangingAction(this, globalCtx) == 0) {
+        EnIk_DetermineAttack(this);
+    }
+}
+
+void EnIk_SetupVerticalAttack(EnIk* this) {
+    s32 pad;
+    f32 playbackSpeed;
+
+    this->actor.speedXZ = 0.0f;
+    if (this->drawArmorFlags != 0) {
+        playbackSpeed = 1.5f;
+    } else {
+        playbackSpeed = 1.2f;
+    }
+    Animation_Change(&this->skelAnime, &gIronKnuckleVerticalAttackAnim, playbackSpeed, 0.0f,
+                     Animation_GetLastFrame(&gIronKnuckleVerticalAttackAnim.common), 3, -4.0f);
+    this->counter = 0;
+    this->unk_2F8 = -1;
+    this->actionFunc = EnIk_VerticalAttack;
+}
+
+void EnIk_VerticalAttack(EnIk* this, GlobalContext* globalCtx) {
+    Vec3f sp2C;
+
+    this->counter++;
+    if (!(this->skelAnime.curFrame < 7.0f) || !EnIk_IsChangingAction(this, globalCtx)) {
+        if (Animation_OnFrame(&this->skelAnime, 15.0f)) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_SWING_AXE);
+
+        } else if (Animation_OnFrame(&this->skelAnime, 21.0f) != 0) {
+            sp2C.x = (Math_SinS((this->actor.shape.rot.y + 0x6A4)) * 70.0f) + this->actor.world.pos.x;
+            sp2C.z = (Math_CosS((this->actor.shape.rot.y + 0x6A4)) * 70.0f) + this->actor.world.pos.z;
+            sp2C.y = this->actor.world.pos.y;
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_HIT_GND);
+            func_800DFD04(globalCtx->cameraPtrs[globalCtx->activeCamera], 2, 25, 5);
+            func_8013ECE0(this->actor.xyzDistToPlayerSq, 180, 20, 100);
+            CollisionCheck_SpawnShieldParticles(globalCtx, &sp2C);
+        }
+
+        if ((this->skelAnime.curFrame > 13.0f) && (this->skelAnime.curFrame < 23.0f)) {
+            this->colliderQuad.base.atFlags |= AT_ON;
+            if (this->drawArmorFlags != 0) {
+                this->actor.speedXZ = sin_rad((this->skelAnime.curFrame - 13.0f) * (M_PI / 20)) * 10.0f;
+            }
+        } else {
+            this->colliderQuad.base.atFlags &= ~AT_ON;
+            this->actor.speedXZ = 0.0f;
+        }
+
+        if ((this->drawArmorFlags != 0) && (this->skelAnime.curFrame < 13.0f)) {
+            Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x5DC);
+            this->actor.world.rot.y = this->actor.shape.rot.y;
+        }
+
+        if (SkelAnime_Update(&this->skelAnime)) {
+            EnIk_SetupPulloutAxe(this);
+        }
+    }
+}
+
+void EnIk_SetupPulloutAxe(EnIk* this) {
+    Animation_PlayOnce(&this->skelAnime, &gIronKnuckleEndVerticalAttackAnim);
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_PULLOUT);
+    this->actionFunc = EnIk_PulloutAxe;
+}
+
+void EnIk_PulloutAxe(EnIk* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion
void EnIk_PullOutAxe(EnIk* this, GlobalContext* globalCtx) {
```
Or TakeOutAxe. (I suppose we should rule out DrawAxe...)